hazelcast/include/hazelcast/client/FlakeIdGenerator.h

```diff
@@ -0,0 +1,95 @@
+         * millisecond and uniqueness is guaranteed in this case. The implementation does not allow overflowing
+         * by more than 15 seconds, if IDs are requested at higher rate, the call will block. Note, however, that
+         * clients are able to generate even faster because each call goes to a different (random) member and
+         * the 64 IDs/ms limit is for single member.
+         *
+         * <h4>Node ID overflow</h4>
+         * Node ID component of the ID has 16 bits. Members with member list join version higher than
+         * 2^16 won't be able to generate IDs, but functionality will be preserved by forwarding to another
+         * member. It is possible to generate IDs on any member or client as long as there is at least one
+         * member with join version smaller than 2^16 in the cluster. The remedy is to restart the cluster:
+         * nodeId will be assigned from zero again. Uniqueness after the restart will be preserved thanks to
+         * the timestamp component.
+         *
+         * @since 3.10.1
+         */
+    class HAZELCAST_API FlakeIdGenerator : public IdGenerator {
+            friend class impl::HazelcastClientInstanceImpl;
+        public:
+            /**
+             * Generates and returns a cluster-wide unique ID.
+             * <p>
+             * Operation on member is always local, if the member has valid node ID, otherwise it's remote. On
+             * client, this method goes to a random member and gets a batch of IDs, which will then be returned
+             * locally for limited time. The pre-fetch size and the validity time can be configured for
+             * each client and member, see {@code ClientConfig.addFlakeIdGeneratorConfig()} for client config.
+             * <p>
+             * <b>Note:</b> Values returned from this method may be not strictly ordered.
+             *
+             * @return new cluster-wide unique ID
+             *
+             * @throws NodeIdOutOfRangeException if node ID for all members in the cluster is out of valid range.
+             *      See "Node ID overflow" in {@link FlakeIdGenerator class documentation} for more details.
+             * @throws UnsupportedOperationException if the cluster version is below 3.10
+             */
+            virtual int64_t newId();
+
+            /**
+             * {@inheritDoc}
+             *
+             * @deprecated
+             *
+             * <p>
+             * <b>Note for Flake ID Generator:</b> This method does nothing and will simply tell if the next ID
+             * will be larger than the given ID.
+             * You don't need to call this method on cluster restart - uniqueness is preserved thanks to the
+             * timestamp component of the ID.
+             *
+             * @return {@code true}, if the next ID will be larger than the supplied id
+             */
+            virtual bool init(int64_t id);
```

Comment:
Why do we have this method if it is already deprecated in Java?

Reply:
I guess that is because you implement `IdGenerator`

Reply:
Yes, it is there for inheritence from IdGenerator. We omitted this until the last commit where we decided that it is better to keep the inheritence in place.